soroban-env-host/src/native_contract/token/admin.rs

```diff
@@ -20,29 +18,12 @@ pub fn write_administrator(e: &Host, id: Identifier) -> Result<(), Error> {
 
 pub fn check_admin(e: &Host, auth: &Signature) -> Result<(), Error> {
     let admin = read_administrator(e)?;
-    match (admin, auth) {
-        (Identifier::Contract(admin_id), Signature::Contract) => {
-            let invoker_id: BytesN<32> = e.get_invoking_contract()?.to_raw().try_into_val(e)?;
-            if admin_id.compare(&invoker_id)? != Ordering::Equal {
-                Err(Error::ContractError)
-            } else {
-                Ok(())
-            }
-        }
-        (Identifier::Ed25519(admin_id), Signature::Ed25519(signature)) => {
-            if admin_id.compare(&signature.public_key)? != Ordering::Equal {
-                Err(Error::ContractError)
-            } else {
-                Ok(())
-            }
-        }
-        (Identifier::Account(admin_id), Signature::Account(signatures)) => {
-            if admin_id.compare(&signatures.account_id)? != Ordering::Equal {
-                Err(Error::ContractError)
-            } else {
-                Ok(())
-            }
-        }
-        _ => Err(Error::ContractError),
+
+    let id = Signature::get_identifier(auth, e)?;
```

Comment:
Is there a reason you can't do this here?
```suggestion
    let id = auth.get_identifier(e)?;
```
Just curious.